akka-persistence-testkit/src/main/scala/akka/persistence/testkit/scaladsl/PersistenceTestKit.scala

```diff
@@ -0,0 +1,264 @@
+
+  def expectPersistedInAnyOrder[A](persistenceId: String, msgs: immutable.Seq[A]): immutable.Seq[A]
+
+  def expectPersistedInAnyOrder[A](persistenceId: String, msgs: immutable.Seq[A], max: FiniteDuration): immutable.Seq[A]
+
+  def rejectNextNPersisted(persistenceId: String, n: Int): Unit
+
+  def rejectNextPersisted(persistenceId: String): Unit = rejectNextNPersisted(persistenceId, 1)
+
+  def rejectNextNPersisted(n: Int): Unit
+
+  def rejectNextPersisted(): Unit = rejectNextNPersisted(1)
+
+  def rejectNextRead(): Unit = rejectNextNReads(1)
+
+  def rejectNextNReads(n: Int): Unit
+
+  def rejectNextRead(persistenceId: String): Unit = rejectNextNReads(persistenceId, 1)
+
+  def rejectNextNReads(persistenceId: String, n: Int): Unit
+
+  def rejectNextDelete(): Unit = rejectNextNDeletes(1)
+
+  def rejectNextNDeletes(n: Int): Unit
+
+  def rejectNextDelete(persistenceId: String): Unit = rejectNextNDeletes(persistenceId, 1)
+
+  def rejectNextNDeletes(persistenceId: String, n: Int): Unit
+
+}
+
+class SnapshotTestKit(override val storage: TestKitStorage[(SnapshotMetadata, Any), SnapshotOperation])(implicit val system: ActorSystem)
+  extends CommonTestKitOps[(SnapshotMetadata, Any), SnapshotOperation]
+  with PolicyOpsTestKit[SnapshotOperation]
+  with ExpectOps[(SnapshotMetadata, Any)]
+  with HasStorage[(SnapshotMetadata, Any), SnapshotOperation] {
+  require(Try(Persistence(system).journalFor(PersistenceTestKitSnapshotPlugin.PluginId)).isSuccess, "The test persistence plugin for snapshots is not configured")
+
+  import SnapShotStorageEmulator._
+  import SnapshotTestKit._
+
+  private val settings = SettingsExtension(system)
+
+  override private[testkit] val pollInterval: FiniteDuration = settings.pollInterval
+
+  override private[testkit] val maxTimeout: FiniteDuration = settings.assertTimeout
+
+  override private[testkit] val Policies = SnapShotStorageEmulator.SnapshotPolicies
+
+  override def failNextNPersisted(persistenceId: String, n: Int): Unit =
+    failNextNOpsCond((pid, op) ⇒ pid == persistenceId && op.isInstanceOf[Write], n)
+
+  override def failNextNPersisted(n: Int): Unit =
+    failNextNOpsCond((_, op) ⇒ op.isInstanceOf[Write], n)
+
+  override def failNextNReads(n: Int): Unit =
+    failNextNOpsCond((_, op) ⇒ op.isInstanceOf[Read], n)
+
+  override def failNextNReads(persistenceId: String, n: Int): Unit =
+    failNextNOpsCond((pid, op) ⇒ pid == persistenceId && op.isInstanceOf[Read], n)
+
+  override def failNextNDeletes(n: Int): Unit =
+    failNextNOpsCond((_, op) ⇒ op.isInstanceOf[Delete], n)
+
+  override def failNextNDeletes(persistenceId: String, n: Int): Unit =
+    failNextNOpsCond((pid, op) ⇒ pid == persistenceId && op.isInstanceOf[Delete], n)
+
+  override private[testkit] def reprToAny(repr: (SnapshotMetadata, Any)) = repr._2
+
+}
+
+object SnapshotTestKit {
+
+  object SettingsExtension extends ExtensionId[Settings] {
```

Comment:
I usually give this object the same name as the extension instance class, i.e. s/SettingsExtension/Settings/